### PR TITLE
fix(api): Match the deck definition to the robot type in protocol analysis

### PR DESCRIPTION
**Pull request by SyntaxColoring**
# Overview

This fixes a bunch of places that were using bad global defaults for the deck type. They now choose the deck type according to the protocol's `robotType`, instead.

Closes RSS-156.

# Test plan

* [x] Run any OT-2 protocol on an OT-2 and make sure the pipettes continue to move to the correct place. Slot 1 doesn't count because it's the same across robots.
* [x] Run any OT-3 protocol on an OT-3 and make sure the pipettes continue to move to the correct place. Slot 1 doesn't count because it's the same across robots.
* [x] While doing that, make sure in-app and on-robot analysis aren't throwing any new errors.

This PR includes a new integration test for in-app analysis.

# Changelog

This PR finds the places that look to environment variables to determine the deck definition, and changes them to look at the protocol's declared `robotType` instead. After #11812, identifying these places should be easy: they should be all the places that call `deck_type.guess_from_global_config()`.

# Review requests

Grep our codebase for any remaining uses of `deck_type.guess_from_global_config()`. Make sure none of these are surprising.

I know there's one problematic one in the code for `opentrons_simulate` / `opentrons.simulate.simulate()` / `opentrons.simulate.get_protocol_api()`. I think it's easiest if we address that as part of RCORE-535.

# Risk assessment

Medium. We're undoing long-held assumptions that there is "a" "current deck." This is mostly just for simulation, but some of the code is shared and affects actual execution, so we need to be careful.

### CODE REVIEW

File: api/src/opentrons/protocol_engine/resources/deck_data_provider.py [hunk 1]
```diff
@@ -30,23 +29,27 @@ class DeckDataProvider:
...
-    # NOTE(mc, 2020-11-18): async to allow file reading and parsing to be
-    # async on a worker thread in the future
-    @staticmethod
-    async def get_deck_definition() -> DeckDefinitionV3:
+    async def get_deck_definition(self) -> DeckDefinitionV3:
         """Get a labware definition given the labware's identification."""
-        return load_deck(guess_deck_type_from_global_config(), 3)
+
+        def sync() -> DeckDefinitionV3:
+            return load_deck(self._deck_type.value, 3)
```
Comment (TamarZanzouri) [inline api/src/opentrons/protocol_engine/resources/deck_data_provider.py]: not a big deal but can we use a variable to the schema version or at least add the arg name or use the default value? had to dig in to see what is the 3 for

File: api/tests/opentrons/cli/test_cli.py [hunk 2]
```diff
@@ -16,25 +18,112 @@ def _list_fixtures(version: int) -> Iterator[Path]:
     )
 
 
+def _get_analysis_result(protocol_files: List[Path]) -> Tuple[int, Any]:
+    """Run `protocol_files` as a single protocol through the analysis CLI.
```
Comment (TamarZanzouri) [inline api/tests/opentrons/cli/test_cli.py:22]: Nice!

File: api/tests/opentrons/cli/test_cli.py [hunk 3]
```diff
@@ -16,25 +18,112 @@ def _list_fixtures(version: int) -> Iterator[Path]:
...
+        ("2.14", "OT-2", "(196.38, 42.785, 44.04)"),
+        pytest.param(
+            "2.14",
+            "OT-3",
+            "(227.88, 42.785, 44.04)",
+            marks=pytest.mark.ot3_only,  # Analyzing an OT-3 protocol requires an OT-3 hardware API.
+        ),
+    ],
+)
+def test_analysis_deck_definition(
```
Comment (TamarZanzouri) [inline api/tests/opentrons/cli/test_cli.py:100]: nice!

**Review by TamarZanzouri — APPROVED** [review]
code changes look great! left one comment if we can accommodate that. Pending on hardware testing of course

**Review by jbleon95 — APPROVED** [review]
LGTM, tested a simple protocol that picked up a tip and moved to a well, on edge and on this branch with identical offsets, on OT-2 and Flex, all worked as expected. No issues seen in analysis.

**Pull request merged**